internal/listtoken/item.go

```diff
@@ -1,20 +1,19 @@
 // Copyright (c) HashiCorp, Inc.
 // SPDX-License-Identifier: BUSL-1.1
 
-package refreshtoken
+package listtoken
 
 import (
-	"time"
-
 	"github.com/hashicorp/boundary/internal/db/timestamp"
 	"github.com/hashicorp/boundary/internal/types/resource"
 )
 
-// Item represents a generic resource with a public ID, update time
-// and resource type.
+// Item represents a generic resource with a public ID,
+// create time, update time and resource type.
 type Item struct {
 	publicId     string
-	updateTime   time.Time
+	createTime   *timestamp.Timestamp
+	updateTime   *timestamp.Timestamp
```

Comment:
Create Time and Update Time are never set together on this type. When we have a `PaginationToken` component we set the create time, and when we have a `RefreshToken` component we set the update time. I don't _love_ this, but it _is_ really nice to be able to use this alongside one of the `boundary.Resource` types. In practice the service list method validation will ensure we use the right list token component in the right place.

Reply:
Good ideas!

Reply:
Done